packages/generate-persisted-query-manifest/cli.js

```diff
@@ -1,10 +1,46 @@
 #!/usr/bin/env node
 
-const { generatePersistedQueryManifest } = require("./dist/index.js");
+const { Command } = require("commander");
+const { cosmiconfig } = require("cosmiconfig");
+const { generatePersistedQueryManifest, defaults } = require("./dist/index.js");
+const { TypeScriptLoader } = require("cosmiconfig-typescript-loader");
+const { version } = require("./package.json");
+const { writeFileSync } = require("node:fs");
 
-generatePersistedQueryManifest()
-  .then((manifest) => console.log(JSON.stringify(manifest, null, 2)))
-  .catch((e) => {
-    console.error(e);
-    process.exit(1);
+const program = new Command();
+
+const moduleName = "persisted-query-manifest";
+const supportedExtensions = ["json", "yml", "yaml", "js", "ts", "cjs"];
+
+const searchPlaces = supportedExtensions.flatMap((extension) => [
+  `.${moduleName}.config.${extension}`,
```

Comment:
Prefixing with the dot would make it a hidden file and some may prefer that. I don't think its a bad thing to support both, especially because cosmiconfig does this [out of the box](https://github.com/cosmiconfig/cosmiconfig#explorersearch). I'm sure we could find a way to make documentation succinct. I'm just not driven by "too much to document" as being the reason to limit the options here.

The reason I have `searchPlaces` to begin with and I don't just use `cosmiconfig('persisted-query-manifest')` is because the manifest file is written to `persisted-query-manifest.json`, which is included in list of files that cosmiconfig will look by default. To differentiate, I figured I'd enforce `.config` in the filename.

Reply:
OK with an agree to disagree here too. I do think that letting people specify their own config filename means we can have a more opinionated default, but if your opinion is to have a vaguer opinion then I guess it's opinionated to be less opinionated :)

Reply:
> we can have a more opinionated default

I'm not necessarily against this. In fact, I'd say the search places I have here now are actually a bit more opinionated than the cosmiconfig defaults (I enforce `.config.<ext>`. cosmiconfig will search for `rc` files as well, etc).

In fact I could probably simplify how we look for this by saying:

> Create a config file named `persisted-query-manifest.config.<extension>` where `<extension>` is one of `json`, `yml`, `yaml`, `js`, `ts`, or `cjs`. You may also prefix the file with a dot (`.`) if you prefer hidden files.

What would you like to see as a more opinionated default?

Reply:
I do think that's a better way to document it than a long list. I guess my opinion would be to drop the final sentence and people who want hidden files are welcome to pass `--config`. But it's bikeshed-painting at this point.

Reply:
True, but the reason I left it is because client teams are typically used to config files that support `.` in the name (jest, prettier, eslint, etc).

Agreed though, just bikeshedding at this point 😆